src/wallet/load.cpp

```diff
@@ -22,7 +22,7 @@ bool VerifyWallets(interfaces::Chain& chain)
         fs::path wallet_dir = gArgs.GetArg("-walletdir", "");
         boost::system::error_code error;
         // The canonical path cleans the path, preventing >1 Berkeley environment instances for the same directory
-        fs::path canonical_wallet_dir = fs::canonical(wallet_dir, error);
+        fs::path canonical_wallet_dir = fs::canonical(wallet_dir, error).remove_trailing_separator();
```

Comment:
Why this change is required? I see no broken tests on the current 22.x branch (9b5f674abb0e90280436fbacb8711250cde11773) on macOS 12.2.1 (Apple M1).

Reply:
On my system with boost 1.78, the unit tests fail with a trailing backslash in the path here.